hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/utils/SASGenerator.java

```diff
@@ -85,6 +88,22 @@ private void initializeMac() {
     }
   }
 
+  protected String getCanonicalAccountName(String accountName) throws InvalidConfigurationValueException {
+    // returns the account name without the endpoint
+    int dotIndex = accountName.indexOf(AbfsHttpConstants.DOT);
```

Comment:
please add in comments, what is the semantic of string it is going to operate on.

Reply:
Sorry, it is unclear to me what exactly you are asking me to comment about. Can you please elaborate?

Reply:
You have some string computation over accountname. It would be great if you could share the semantics of accountname. For example, is it accountname.dfs.azure.com.

Hope things like containerName@accountname.dfs.azure.com can't happen.

Thanks.